### ISSUE TITLE
Could not install packages

### ISSUE THREAD

**Issue by xgenem** [mattermost/mattermost-mobile#6329]
#### Summary
`npm install` failed.

### Environment Information
  - Device Name: Macbook Pro M1
  - OS Version: Monterey 12.3.1
  - Mattermost App Version: n/a
  - Mattermost Server Version: n/a

#### Steps to reproduce
Clone latest github and `npm install`

#### Expected behavior
Expected behaviour is to install the packages without issues

#### Observed behavior (that appears unintentional)
```npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: @rudderstack/rudder-sdk-react-native@1.1.1
npm ERR! Found: react-native@0.67.2
npm ERR! node_modules/react-native
npm ERR!   react-native@"0.67.2" from the root project
npm ERR!   peer react-native@"*" from @mattermost/react-native-paste-input@0.3.6
npm ERR!   node_modules/@mattermost/react-native-paste-input
npm ERR!     @mattermost/react-native-paste-input@"0.3.6" from the root project
npm ERR!   44 more (@react-native-async-storage/async-storage, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react-native@"0.41.2" from @rudderstack/rudder-sdk-react-native@1.1.1
npm ERR! node_modules/@rudderstack/rudder-sdk-react-native
npm ERR!   @rudderstack/rudder-sdk-react-native@"1.1.1" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: react-native@0.41.2
npm ERR! node_modules/react-native
npm ERR!   peer react-native@"0.41.2" from @rudderstack/rudder-sdk-react-native@1.1.1
npm ERR!   node_modules/@rudderstack/rudder-sdk-react-native
npm ERR!     @rudderstack/rudder-sdk-react-native@"1.1.1" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /Users/admin/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/admin/.npm/_logs/2022-05-31T22_14_22_061Z-debug-0.log
```

#### Possible fixes
N/A

**Comment by xgenem** [comment]
Installed with `--legacy-peer-deps` flag but I don't think this is an acceptable solution.

**Comment by enahum** [comment]
What is your node and npm versions?

**Comment by xgenem** [comment]
Hi @enahum 
```
$ node -v
v16.13.1
$ npm -v
8.10.0
```

Cheers,

**Comment by enahum** [comment]
> Hi @enahum 
> 
> ```
> 
> $ node -v
> 
> v16.13.1
> 
> $ npm -v
> 
> 8.10.0
> 
> ```
> 
> 
> 
> Cheers,

Downgrade to npm 8.5.5 and it should be fine

**Comment by ElonVolo** [comment]
@enahum 

I had the same issue when I tried to install and run the mattersmost mobile project today and the downgraded npm seemed to work fine.

One suggestion I have for preventing confusion and problems over incompatible versions of global binaries (such as those in the issue) would be to incorporate an environment checker such as  [Solidarity](https://infinitered.github.io/solidarity) into the package.json pre-install script, which would check to make sure that all the necessary command line global binaries versions (npm, react native cli, etc), Xcode/Android Studio versions, and environment variables are all configured correctly. If  anything's missing or is installed at the wrong version it would immediately inform the new users of the corrections that would need to be made and then package installation would be halted.

I've forked off mattermost and have created a poc of how this might work.

https://github.com/ElonVolo/mattermost-mobile/tree/gekidou-solidarity-poc

Trying cloning the fork and checking out the __gekidou-solidarity-poc__ branch and then changing the __NPM__ entry in the __.solidarity__ config file from:

```
    "NPM": [{ "rule": "cli", "binary": "npm", "semver": "8.5.5" }],
```

to something incompatible with the current project, such at 8.10.0

```
    "NPM": [{ "rule": "cli", "binary": "npm", "semver": "8.10.0" }],
```

You'll then get output similar to the image below the informs the developer that they can't go any further.

<img width="658" alt="Screen Shot 2022-06-10 at 3 07 43 AM" src="https://user-images.githubusercontent.com/63563814/173010063-38db2ba1-a45b-4cc8-99c2-f5cebaaeb3be.png">
